packages/sparrowql-graphql/src/graph.ts

```diff
@@ -67,13 +67,13 @@ export function astToOptions(
           local: null,
           // TODO: Typings do not match for NamedType
           // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-unsafe-member-access
-          to: type2collection[(node.type as any).type.name.value as string],
+          to: type2collection[node.type.type.name.value],
```

Comment:
This is the most problematic one. `node.type` is of type `TypeNode`.
```ts
export type TypeNode = NamedTypeNode | ListTypeNode | NonNullTypeNode;

export interface NamedTypeNode {
  readonly kind: 'NamedType';
  readonly loc?: Location;
  readonly name: NameNode;
}

export interface ListTypeNode {
  readonly kind: 'ListType';
  readonly loc?: Location;
  readonly type: TypeNode;
}

export interface NonNullTypeNode {
  readonly kind: 'NonNullType';
  readonly loc?: Location;
  readonly type: NamedTypeNode | ListTypeNode;
}
```

We expect `node.type` to be either of type `ListTypeNode | NonNullTypeNode` and then we expect their `.type` to be exactly `NamedTypeNode` since this is the only property that has a `name.value` property. I could write a typeguard for this but I'm not sure what kind of behavior do we expect if it doesn't pass - throw an error? Early return?

Reply:
Please check, whether [this test](https://github.com/vazco/sparrowql/blob/3fb1c878d2053dfd7c6085cb84aac1194a3d968b/__tests__/demo-graphql.js#L27-L54) works without `!` on fields marked with `@relationTo`. I guess it won't, and that's a great catch. We'd need a function, like `resolveTypeName(type: TypeNode): string` and use it here.

Reply:
It's just as you predicted, the test failed.

```
 FAIL  __tests__/demo-graphql.js
  ● demo

    TypeError: Cannot read property 'name' of undefined

      37 |                     // TODO: Typings do not match for NamedType
      38 |                     // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-unsafe-member-a
ccess
    > 39 |                     to: type2collection[node.type.type.name.value],
         |                                                        ^
      40 |                 };
      41 |                 for (const { name, value } of (_a = directive.arguments) !== null && _a !== void 0 ? _a : []) {
      42 |                     // Typings do not match for VariableNode

      at Object.name [as FieldDefinition] (packages/sparrowql-graphql/lib/graph.js:39:56)
      at Object.visit (packages/sparrowql-graphql/node_modules/graphql/language/visitor.js:243:26)
      at visit (packages/sparrowql-graphql/lib/graph.js:23:16)
      at astToOptions (__tests__/demo-graphql.js:56:21)
      at Object.<anonymous> (__mongo__/environment.js:19:21)
```

After adding the following function everything passes. Is this an acceptable solution?

```
function resolveTypeName(node: TypeNode): string {
  if (isNamedTypeNode(node)) {
    return node.name.value;
  }
  return resolveTypeName(node);
}
```

I will also add a test to cover this case.

Reply:
Yep, it looks fine.